msim/src/sim/net/mod.rs

```diff
@@ -905,6 +905,7 @@ impl NetSim {
         debug!("delete_node {id}");
         let mut network = self.network.lock().unwrap();
         network.delete_node(id);
+        drop(network);
```

Comment:
Why this line?

Reply:
must release the lock before deleting the node from host_state below - that can call drop handlers which try to acquire this lock, which causes deadlocks.